### PR TITLE
fds icon now uses currentColor

### PR DESCRIPTION
**Pull request by rey-wright**
## Packages changed

- [ ] `fds-dictionary`
- [x] `fds-styles`
- [x] `fds-components`
- [ ] `fds-mui-theme`
- [x] `fds-icons`

## Description
`.fds-icon` now uses `currentColor` under the hood. I reverted some usage of currentColor in `fds-components` that is no longer needed with this change, and this class is heavily used inside `fds-icons` package as well.

## Checklist
- [X] Docs have been rebuilt (`yarn build:full`)
- [X] All unit tests pass
- [X] Snapshots have been updated
- [X] No lint errors or warnings have been introduced
- [X] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

**Comment by rey-wright** [comment]
I could use some help testing this via `yarn link`... I wasn't totally able to test this... I may be missing something simple though.

File: packages/fds-components/src/style/interactive/buttonGroup.css [hunk 1]
```diff
@@ -43,11 +43,6 @@
     border var(--button-group-timeout) var(--motion-easing-default);
 }
 
-.groupbtn .fds-icon {
```
Comment (akdetrick) [inline packages/fds-components/src/style/interactive/buttonGroup.css]: so unfortunately we can't remove this style until `fds-components` is updated to the `2.33` of `fds-icons` in a separate PR.

Comment (rey-wright) [reply packages/fds-components/src/style/interactive/buttonGroup.css]: got it. I know you explained this to me... I feel like I understood this last week, but need to wrap my head around it again. Reverting!

**Comment by rey-wright** [comment]
Thanks for the review! I'll try to handle the bumping of the components library.

**Pull request merged**